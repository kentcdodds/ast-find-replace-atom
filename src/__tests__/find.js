import path from 'path'

const find = require('../find')

const fixture = n => require.resolve(`./fixtures/${n}`)

test('finds nothing', () => {
  const result = find(fixture('nothing'))
  expect(result).toEqual([])
})

test('finds something', () => {
  const results = find(fixture('something'))
  expect(relativizePaths(results)).toMatchSnapshot()
})

/**
 * This takes the results object and removes environment-specific
 * elements from the path.
 * @param {Object} results - This is the results object from find
 * @return {Object} - The new results object with the clean paths
 */
function relativizePaths(results) {
  return Object.keys(results).reduce((obj, key) => {
    const newKey = key
      .replace(':/', ':\\')
      .replace(path.resolve(__dirname, '../..'), '<projectRootDir>')
      .replace(/\\/g, '/')
    obj[newKey] = results[key]
    return obj
  }, {})
}
