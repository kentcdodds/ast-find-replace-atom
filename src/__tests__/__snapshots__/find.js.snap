// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`finds something 1`] = `
Object {
  "<projectRootDir>/src/__tests__/fixtures/findables/file-1.js": Array [
    SourceLocation {
      "end": Position {
        "column": 7,
        "line": 1,
      },
      "identifierName": "console",
      "start": Position {
        "column": 0,
        "line": 1,
      },
    },
  ],
}
`;
